server/app/dal/Dao.scala

```diff
@@ -62,6 +72,11 @@ class DaoImpl @Inject()(cluster: CassandraCluster)(implicit ec: ExecutionContext
       val limitInt: Integer = limit
       session.executeAsync(getLastTasksStatement.bind(year(yearAgo), `type`.toString, limitInt))
     }.map(allTasks))
+
+  def getTask(year: Long, `type`: TaskType, timeuuid: UUID): Future[Option[Task]] = TryFuture(
+    toFuture {
+      session.executeAsync(getTaskStatement.bind(new Timestamp(year), `type`.toString, timeuuid))
```

Comment:
Sometimes Dao may wrap values with necessary transport types and which is ok and does not bring extra responsibility to the Dao itself. But I agree such things should be minimized as much as possible.